app/controllers/forest_liana/base_controller.rb

```diff
@@ -2,5 +2,31 @@ module ForestLiana
   class BaseController < ::ActionController::Base
     skip_before_action :verify_authenticity_token, raise: false
     wrap_parameters false
+    before_action :reject_unauthorized_ip
+
+    private
+
+    def reject_unauthorized_ip
+      begin
+        ip = request.remote_ip
+
+        if !IpWhitelist.is_ip_whitelist_retrieved || !IpWhitelist.is_ip_valid(ip)
+          unless IpWhitelist.retrieve
+            raise Errors::HTTP403Error.new("IP whitelist not retrieved")
+          end
+
+          unless IpWhitelist.is_ip_valid(ip)
+            raise Errors::HTTP403Error.new("IP address rejected (#{ip})")
+          end
+        end
+      rescue => exception
```

Comment:
If you want to rescue `Errors::HTTP403Error` only you can:
`rescue Errors::HTTP403Error => exception`

Other exceptions should be handled differently as `display_error` is not a function implemented for any error object. With another generic `rescue => exception` below the first one.